app/jobs/statcan_sync_job.rb

```diff
@@ -0,0 +1,13 @@
+class StatcanSyncJob < ApplicationJob
+  queue_as :default
+
+  def perform(statcan_dataset_id)
+    dataset = StatcanDataset.find(statcan_dataset_id)
+    data = StatcanFetcher.fetch(dataset.statcan_url)
```

Comment:
nit: Rails has a convention to prefer fat models and skinny controllers (and imo that applies to jobs too). 

I'd rather have a method on a `StatcanDataset` called `sync!` instead of having the logic for refreshing live in the job itself.

Combining this and the above suggestion you get:

```
def perform(statcan_dataset)
    statcan_dataset.sync!
end
```

Reply:
7eb00ab1e2b9370f280185bdfa89f6e30282937c